flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java

```diff
@@ -16,15 +16,253 @@
+                }
+            } else {
+                // Convert to Dynamic Message.
+                value = toProtoValue(fieldDescriptor, field.schema(), value);
+                builder.setField(fieldDescriptor, value);
+            }
+        }
+        return builder.build();
+    }
+
+    /**
+     * Function to convert a value of an AvroSchemaField value to required Dynamic Message value.
+     *
+     * @param fieldDescriptor {@link com.google.protobuf.Descriptors.FieldDescriptor} Object
+     *     describing the sink table field to which given value needs to be converted to.
+     * @param avroSchema {@link Schema} Object describing the value of Avro Schema Field.
+     * @param value Value of the Avro Schema Field.
+     * @return Converted Object.
+     */
+    private static Object toProtoValue(
+            FieldDescriptor fieldDescriptor, Schema avroSchema, Object value) {
+        switch (avroSchema.getType()) {
+            case RECORD:
+                return getDynamicMessageFromGenericRecord(
+                        (GenericRecord) value, fieldDescriptor.getMessageType());
+            case ARRAY:
+                Iterable<Object> iterable = (Iterable<Object>) value;
+                // Get the inner element type.
+                @Nullable Schema arrayElementType = avroSchema.getElementType();
+                if (arrayElementType.isNullable()) {
+                    throw new IllegalArgumentException("Array cannot have NULLABLE datatype");
+                }
+                if (arrayElementType.isUnion()) {
+                    throw new IllegalArgumentException(
+                            "ARRAY cannot have multiple datatypes in BigQuery.");
+                }
+                // Convert each value one by one.
+                return StreamSupport.stream(iterable.spliterator(), false)
+                        .map(v -> toProtoValue(fieldDescriptor, arrayElementType, v))
+                        .collect(Collectors.toList());
+            case UNION:
+                Schema type = AvroSchemaHandler.handleUnionSchema(avroSchema).getLeft();
+                // Get the schema of the field.
+                return toProtoValue(fieldDescriptor, type, value);
+            case MAP:
+                throw new UnsupportedOperationException("Not supported yet");
+            case STRING:
+                /*
+                Logical Types currently supported are of the following underlying types:
+                    DATE - INT
+                    DECIMAL - BYTES
+                    TIMESTAMP-MICROS - LONG
+                    TIMESTAMP-MILLIS - LONG
+                    UUID - STRING
+                    TIME-MILLIS - INT/STRING
+                    TIME-MICROS - LONG/STRING
+                    LOCAL-TIMESTAMP-MICROS - LONG/STRING
+                    LOCAL-TIMESTAMP-MILLIS - LONG/STRING
+                    GEOGRAPHY - STRING
+                    JSON - STRING
+                */
+            case LONG:
```

Comment:
same as `INT`, if there is a logical type it would be handled (logical type handling in the next [PR](https://github.com/GoogleCloudDataproc/flink-bigquery-connector/pull/101)), otherwise the value passed would be returned as it is.

Reply:
i'm not very familiar with java switch statements but does the string and long case fall through to the int case or does it fall through to the default case? either way, let's add a comment

Reply:
Added!

Reply:
thanks; you can resolve this!